spark/v3.0/spark/src/main/java/org/apache/iceberg/spark/actions/Spark3BinPackStrategy.java

```diff
@@ -61,9 +62,17 @@ public Table table() {
       SparkSession cloneSession = spark.cloneSession();
       cloneSession.conf().set(SQLConf.ADAPTIVE_EXECUTION_ENABLED().key(), false);
 
+      long targetReadSize = splitSize(inputFileSize(filesToRewrite));
+      // Ideally this would be the row-group size but the row group size is not guaranteed to be consistent
+      long fileSplitSize = Long.valueOf(table.properties().getOrDefault(
+          TableProperties.PARQUET_ROW_GROUP_SIZE_BYTES,
```

Comment:
It looks a bit odd to simply check the row group config as the table may have both Avro and Parquet and we may be compacting Avro, for example. I know it is something @rdblue has suggested so I'll leave it to you folks.